File: app/src/main/java/com/recursia/popularmovies/domain/MoviesListInteractor.java
package com.recursia.popularmovies.domain;

import com.recursia.popularmovies.domain.models.Movie;

import java.util.List;

import io.reactivex.Single;

public interface MoviesListInteractor {

    Single<List<Movie>> discoverMovies(String sortBy, int page, int voteCount, String language);

}

File: app/src/main/java/com/recursia/popularmovies/domain/MoviesListInteractor.kt
package com.recursia.popularmovies.domain

import com.recursia.popularmovies.domain.models.Movie

import io.reactivex.Single

interface MoviesListInteractor {

    fun discoverMovies(sortBy: String, page: Int, voteCount: Int, language: String): Single<List<Movie>>
}
